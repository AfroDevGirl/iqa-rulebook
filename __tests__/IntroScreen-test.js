import React from 'react'
import IntroScreenView from '../screens/IntroScreen/IntroScreen.js'
import renderer from 'react-test-renderer'

describe('IntroScreenView', () => {
  test('renders correctly', () => {
    const screen = renderer.create(<IntroScreenView />).toJSON()
    expect(screen).toMatchSnapshot()
  })
})
