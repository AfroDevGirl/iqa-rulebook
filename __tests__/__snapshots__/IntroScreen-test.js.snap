// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroScreenView renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <RCTScrollView
          alwaysBounceVertical={true}
          contentContainerStyle={
            Object {
              "backgroundColor": "#fff",
              "paddingVertical": 20,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Introduction
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                The sport of quidditch has recently seen intense international expansion. With 21 teams competing at the IQA Quidditch World Cup 2016, the need for a codified, and singular rulebook through which to regulate international competition, has never been greater.
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                As the sport expands and evolves, it remains the responsibility of the rulebook to maintain pace by addressing fast moving developments in the way the sport is played. The rulebook must be proactive and able to address the safety and gameplay concerns of a sport that continues to evolve at an unprecedented rate, on all continents across the world.
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Quidditch is no longer merely a backyard sport: it can be played in an open field or before an audience of thousands. Whatever the conditions of play, the rulebook needs to have a consistent answer for how the game must work.
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                This rulebook finds its base in USQ rulebook nine, and its core content will be familiar to many experienced players and followers.However, there are some changes that take into account the nature of international play, putting player safety and the integrity of the game as the focal point of the rulebook.
              </Text>
            </View>
            <View
              style={
                Object {
                  "padding": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                For those of you who are new to the sport of quidditch, we recommend that you learn the basic rules from your friends, teammates, and the overview in 1.1. Quidditch: An overview before you read straight through this rulebook. Of course, that is simple a recommendation, you are welcome to dive right in!
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          collapsable={undefined}
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#F7F7F7",
              "borderBottomColor": "#A7A7AA",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={undefined}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 70,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 70,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  accessible={true}
                  allowFontScaling={true}
                  collapsable={undefined}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, .9)",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  Intro Screen
                </Text>
              </View>
              <View
                collapsable={undefined}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits={
                    Array [
                      "button",
                    ]
                  }
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={undefined}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "padding": 8,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Back To Home
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
